### PR TITLE
Version 0.16.0

### PR DESCRIPTION
**Pull request by Kludex**
# Checklist
- [x] Take a decision on #1160, either going with #1165, #1269 or my alternative proposed on the latter.
- [x] #1271
- [x] #1279 
- [x] #1278

### CODE REVIEW

**Comment by Kludex** [comment]
@euri10 Feel free to add items to the checklist.

I'm going to fix what's missing on the PRs that I want, then ask for review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Change Log
 
+## 0.16.0 - 2021-11-27
```
Comment (Kludex) [inline CHANGELOG.md]: Self reminder to change the date!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+- Replace `WSGIResponder.send_queue` type from `list` to `deque`. (#1214) 10/10/21
+
+### Fixed
+
+- Main process exit after startup failure on reloader classes (#1177) 30/09/21
+- Add explicit casting on click options (#1217) 11/10/21
+- Allow WebSocket close event to receive reason being None from ASGI app. (#1259) 23/11/21
+- Fix a bug in `WebSocketProtocol.asgi_receive` on which we returned a close frame even if there were data messages before that frame in the read queue. (#1252) 25/11/21
+- The option `--reload-dirs` was splitting a string into single character directories. (#1267) 25/11/21
+
```
Comment (Kludex) [inline CHANGELOG.md:24]:
```suggestion
- Only second SIGINT is able to forcelly shutdown the server (#1269) 28/11/21

```

**Comment by dimaqq** [comment]
poke...

File: CHANGELOG.md (hunk 2 above)
Comment (dimaqq) [reply CHANGELOG.md:24]:
```suggestion
- Fixed middleware shutdown when server is terminated. (#1160) 10/11/2021
```

**Comment by Kludex** [comment]
@dimaqq We need to wait for someone to review the checklist above. :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,25 @@
 # Change Log
 
+## 0.16.0 - 2021-11-27
+
+### Added
+
+- Bump `websockets` to 10.0. (#1180) 13/09/2021
```
Comment (Kludex) [inline CHANGELOG.md]:
```suggestion
- Enable read of uvicorn settings from environment variables (#1279) 06/12/2021
- Bump `websockets` to 10.0. (#1180) 13/09/2021
- Ensure non-zero exit code when startup fails (#1278) 06/12/2021
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+
+### Fixed
+
+- Main process exit after startup failure on reloader classes (#1177) 30/09/21
+- Add explicit casting on click options (#1217) 11/10/21
+- Allow WebSocket close event to receive reason being None from ASGI app. (#1259) 23/11/21
+- Fix a bug in `WebSocketProtocol.asgi_receive` on which we returned a close frame even if there were data messages before that frame in the read queue. (#1252) 25/11/21
+- The option `--reload-dirs` was splitting a string into single character directories. (#1267) 25/11/21
+- Only second SIGINT is able to forcelly shutdown the server (#1269) 28/11/21
+
```
Comment (Kludex) [inline CHANGELOG.md:24]:
```suggestion
- Allow app-dir parameter on the run() function #1271

```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,28 @@
 # Change Log
 
+## 0.16.0 - 2021-11-27
+
+### Added
+
+- Enable read of uvicorn settings from environment variables (#1279) 06/12/2021
+- Bump `websockets` to 10.0. (#1180) 13/09/2021
+- Ensure non-zero exit code when startup fails (#1278) 06/12/2021
```
Comment (Kludex) [inline CHANGELOG.md]:
```suggestion
- Enable read of uvicorn settings from environment variables (#1279) 06/12/21
- Bump `websockets` to 10.0. (#1180) 13/09/21
- Ensure non-zero exit code when startup fails (#1278) 06/12/21
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,28 @@
...
+
+### Fixed
+
+- Main process exit after startup failure on reloader classes (#1177) 30/09/21
+- Add explicit casting on click options (#1217) 11/10/21
+- Allow WebSocket close event to receive reason being None from ASGI app. (#1259) 23/11/21
+- Fix a bug in `WebSocketProtocol.asgi_receive` on which we returned a close frame even if there were data messages before that frame in the read queue. (#1252) 25/11/21
+- The option `--reload-dirs` was splitting a string into single character directories. (#1267) 25/11/21
+- Only second SIGINT is able to forcelly shutdown the server (#1269) 28/11/21
+- Allow app-dir parameter on the run() function #1271
```
Comment (Kludex) [inline CHANGELOG.md]:
```suggestion
- Allow app-dir parameter on the run() function (#1271) 06/12/21
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,28 @@
 # Change Log
 
+## 0.16.0 - 2021-11-27
```
Comment (Kludex) [inline CHANGELOG.md]:
```suggestion
## 0.16.0 - 2021-12-07
```

**Comment by Kludex** [comment]
Draft release: https://github.com/encode/uvicorn/releases/tag/untagged-7035024477e3f7683b83

@euri10 would you mind reviewing it?

**Review by aminalaee — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,28 @@
 # Change Log
 
+## 0.16.0 - 2021-12-07
```
Comment (Kludex) [inline CHANGELOG.md]:
```suggestion
## 0.16.0 - 2021-12-08
```

**Pull request merged**